Apple Clang 14 build fails with errors

- MacOS 13.2.1 on Apple M1
- Xcode 14.2
- Apple clang version 14.0.0 (clang-1400.0.29.202)

```console
cmake .. -G Xcode
...
cmake --build .
...
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c++ -target arm64-apple-macos13.1 -fmessage-length\=155 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -fcolor-diagnostics -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wshorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions -Wno-implicit-fallthrough -DCMAKE_INTDIR\=\"Debug\" -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk -fstrict-aliasing -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -I/Users/egorodet/git/FTXUI/build/Debug/include -I/Users/egorodet/git/FTXUI/src -isystem /Users/egorodet/git/FTXUI/include -I/Users/egorodet/git/FTXUI/build/build/dom.build/Debug/DerivedSources-normal/arm64 -I/Users/egorodet/git/FTXUI/build/build/dom.build/Debug/DerivedSources/arm64 -I/Users/egorodet/git/FTXUI/build/build/dom.build/Debug/DerivedSources -F/Users/egorodet/git/FTXUI/build/Debug -Wall -Wextra -pedantic -Werror -Wmissing-declarations -Wdeprecated -Wshadow -std\=c++17 -MMD -MT dependencies -MF /Users/egorodet/git/FTXUI/build/build/dom.build/Debug/Objects-normal/arm64/flexbox_helper.d --serialize-diagnostics /Users/egorodet/git/FTXUI/build/build/dom.build/Debug/Objects-normal/arm64/flexbox_helper.dia -c /Users/egorodet/git/FTXUI/src/ftxui/dom/flexbox_helper.cpp -o /Users/egorodet/git/FTXUI/build/build/dom.build/Debug/Objects-normal/arm64/flexbox_helper.o
/Users/egorodet/git/FTXUI/src/ftxui/dom/flexbox_helper.cpp:164:30: error: implicit conversion loses integer precision: 'unsigned long'
      to 'int' [-Werror,-Wshorten-64-to-32]
      for (int i = ys.size() - 1; i >= 0; --i) {  // NOLINT
               ~   ~~~~~~~~~~^~~
/Users/egorodet/git/FTXUI/src/ftxui/dom/flexbox_helper.cpp:175:30: error: implicit conversion loses integer precision: 'unsigned long'
      to 'int' [-Werror,-Wshorten-64-to-32]
      for (int i = ys.size() - 1; i >= 1; --i) {  // NOLINT
               ~   ~~~~~~~~~~^~~
/Users/egorodet/git/FTXUI/src/ftxui/dom/flexbox_helper.cpp:183:30: error: implicit conversion loses integer precision: 'unsigned long'
      to 'int' [-Werror,-Wshorten-64-to-32]
      for (int i = ys.size() - 1; i >= 0; --i) {  // NOLINT
               ~   ~~~~~~~~~~^~~
/Users/egorodet/git/FTXUI/src/ftxui/dom/flexbox_helper.cpp:191:30: error: implicit conversion loses integer precision: 'unsigned long'
      to 'int' [-Werror,-Wshorten-64-to-32]
      for (int i = ys.size() - 1; i >= 0; --i) {  // NOLINT
               ~   ~~~~~~~~~~^~~
4 errors generated.
```

--- a/src/ftxui/component/screen_interactive.cpp
+++ b/src/ftxui/component/screen_interactive.cpp
@@ -174,9 +174,9 @@ void EventListener(std::atomic<bool>* quit, Sender<Task> out) {
     }
 
     const size_t buffer_size = 100;
-    std::array<char, buffer_size> buffer;                     // NOLINT;
-    int l = read(fileno(stdin), buffer.data(), buffer_size);  // NOLINT
-    for (int i = 0; i < l; ++i) {
+    std::array<char, buffer_size> buffer;                        // NOLINT;
+    size_t l = read(fileno(stdin), buffer.data(), buffer_size);  // NOLINT
+    for (size_t i = 0; i < l; ++i) {
       parser.Add(buffer[i]);  // NOLINT
     }
   }

--- a/src/ftxui/dom/flexbox_helper.cpp
+++ b/src/ftxui/dom/flexbox_helper.cpp
@@ -161,7 +161,7 @@ void SetY(Global& g, std::vector<Line> lines) {
     }
 
     case FlexboxConfig::AlignContent::Stretch: {
-      for (int i = ys.size() - 1; i >= 0; --i) {  // NOLINT
+      for (int i = static_cast<int>(ys.size()) - 1; i >= 0; --i) {  // NOLINT
         const int shifted = remaining_space * (i + 0) / (i + 1);
         ys[i] += shifted;
         const int consumed = remaining_space - shifted;
@@ -172,7 +172,7 @@ void SetY(Global& g, std::vector<Line> lines) {
     }
 
     case FlexboxConfig::AlignContent::SpaceBetween: {
-      for (int i = ys.size() - 1; i >= 1; --i) {  // NOLINT
+      for (int i = static_cast<int>(ys.size()) - 1; i >= 1; --i) {  // NOLINT
         ys[i] += remaining_space;
         remaining_space = remaining_space * (i - 1) / i;
       }
@@ -180,7 +180,7 @@ void SetY(Global& g, std::vector<Line> lines) {
     }
 
     case FlexboxConfig::AlignContent::SpaceAround: {
-      for (int i = ys.size() - 1; i >= 0; --i) {  // NOLINT
+      for (int i = static_cast<int>(ys.size()) - 1; i >= 0; --i) {  // NOLINT
         ys[i] += remaining_space * (2 * i + 1) / (2 * i + 2);
         remaining_space = remaining_space * (2 * i) / (2 * i + 2);
       }
@@ -188,7 +188,7 @@ void SetY(Global& g, std::vector<Line> lines) {
     }
 
     case FlexboxConfig::AlignContent::SpaceEvenly: {
-      for (int i = ys.size() - 1; i >= 0; --i) {  // NOLINT
+      for (int i = static_cast<int>(ys.size()) - 1; i >= 0; --i) {  // NOLINT
         ys[i] += remaining_space * (i + 1) / (i + 2);
         remaining_space = remaining_space * (i + 1) / (i + 2);
       }

